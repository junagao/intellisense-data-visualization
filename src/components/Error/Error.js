import React from 'react';
import PropTypes from 'prop-types';

import ErrorDiv from './Error.styles';

const Error = ({ error }) => (
  <ErrorDiv>
    <p>The following unexpected error occurred: {error}.</p>
    <p>Please try again.</p>
  </ErrorDiv>
);

Error.propTypes = {
  error: PropTypes.string.isRequired,
};

export default Error;
