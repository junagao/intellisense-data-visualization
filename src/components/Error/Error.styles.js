import styled from 'styled-components';

const ErrorDiv = styled.div`
  color: #e53e3e;
  font-size: 0.95rem;
  padding: 1.5rem;
  background-color: #fafafa;
  border-radius: 0.25rem;
`;

export default ErrorDiv;
