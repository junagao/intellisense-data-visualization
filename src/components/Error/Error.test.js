import React from 'react';
import { shallow } from 'enzyme';

import Error from './Error';

describe('ErrorMessage', () => {
  let wrapper;
  const error = 'ERROR';

  beforeEach(() => {
    wrapper = shallow(<Error error={error} />);
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one ErrorDiv element', () => {
    expect(wrapper.find('ErrorDiv').length).toEqual(1);
  });

  it('should render two p elements', () => {
    expect(wrapper.find('p').length).toEqual(2);
  });

  it('should render the error message correctly', () => {
    expect(wrapper.find('p').at(0).length).toEqual(1);
    expect(wrapper.find('p').at(0).text()).toEqual(
      `The following unexpected error occurred: ${error}.`,
    );
    expect(wrapper.find('p').at(1).length).toEqual(1);
    expect(wrapper.find('p').at(1).text()).toEqual(`Please try again.`);
  });
});
