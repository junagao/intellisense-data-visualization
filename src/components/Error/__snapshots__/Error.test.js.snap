// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly given the required props 1`] = `
<ErrorDiv>
  <p>
    The following unexpected error occurred: 
    ERROR
    .
  </p>
  <p>
    Please try again.
  </p>
</ErrorDiv>
`;
